lndr-backend/src/Lndr/Handler/Credit.hs

```diff
@@ -294,3 +297,95 @@ twoPartyBalanceHandler p1 p2 currency = do
+    unless (submitterAddress == creditor || submitterAddress == debtor) $
+        throwError (err400 {errBody = "Submitter is not creditor nor debtor."})
+    unless (creditor /= debtor) $
+        throwError (err400 {errBody = "Creditor and debtor cannot be equal."})
+    unless (B.memberR (ucac signedRecord) (lndrUcacAddrs config)) $
+        throwError (err400 {errBody = "Unrecognized UCAC address."})
+
+    -- check that submitter signed the tx
+    signer <- ioEitherToLndr . pure . EU.ecrecover (stripHexPrefix sig) $
+                EU.hashPersonalMessage hash
+    unless (textToAddress signer == submitterAddress) $
+        throwError (err401 {errBody = "Bad submitter sig"})
+
+    currency <- liftIO $ B.lookupR (ucac signedRecord) (lndrUcacAddrs config)
+
+    -- creating function to send notification to notifications api
+    let attemptToNotify notifyAction = do
+            let counterparty = if creditor /= submitterAddress then creditor else debtor
+            pushDataM <- liftIO . withResource pool $ Db.lookupPushDatumByAddress counterparty
+            nicknameM <- liftIO . withResource pool $ Db.lookupNick submitterAddress
+
+            forM_ pushDataM $ \(channelID, platform) -> liftIO $ do
+                responseCode <- sendNotification config (Notification channelID platform nicknameM notifyAction)
+                let logMsg = "Notification response (" ++ T.unpack currency ++ "): " ++ show responseCode
+                pushLogStrLn loggerSet . toLogStr $ logMsg
+
+    -- check if hash is already registered in pending txs
+    pendingCredit <- liftIO . withResource pool $ Db.lookupPending hash
+    case pendingCredit of
+        -- if the submitted credit record has a matching pending record,
+        -- finalize the transaction on the blockchain
+        Just storedRecord -> when (signature storedRecord /= signature signedRecord) $ do
+            let (creditorSig, debtorSig) = if creditor /= submitterAddress
+                                            then (signature storedRecord, signature signedRecord)
+                                            else (signature signedRecord, signature storedRecord)
+
+            finalizeCredit $ BilateralCreditRecord storedRecord creditorSig debtorSig Nothing
+
+            -- send push notification to counterparty
+            when (recordNum == 0) $
+                attemptToNotify CreditConfirmation
+
+        -- if no matching transaction is found, create pending transaction
+        Nothing -> do
+            let processedRecord = calculateSettlementCreditRecord config signedRecord
+            createMultiPendingRecord recordNum pool processedRecord
+            -- send push notification to counterparty
+            when (recordNum == 0) $
+                attemptToNotify NewPendingCredit
+
+    pure NoContent    
+
+
+createMultiPendingRecord :: Int -> Pool Connection -> CreditRecord -> LndrHandler ()
```

Comment:
Now that the design of this change has been settled, DRY this code with the `createPendingRecord` function above. The only difference should be that `unless ((recordNum > 0) || (null existingPending))` check.